internal/nvmeof/controller/controllerserver.go

```diff
@@ -625,21 +654,28 @@ func toRBDMetadataKey(vcKey string) string {
 }
 
 // populateVolumeContext adds NVMe-oF information to volume context for NodeServer.
-func populateVolumeContext(volume *csi.Volume, data *nvmeof.NVMeoFVolumeData) {
+func populateVolumeContext(volume *csi.Volume, data *nvmeof.NVMeoFVolumeData) error {
 	if volume.VolumeContext == nil {
 		volume.VolumeContext = make(map[string]string)
 	}
-	listenerPortStr := strconv.FormatUint(uint64(data.ListenerInfo.Port), 10)
 	gatewayManagementInfoPortStr := strconv.FormatUint(uint64(data.GatewayManagementInfo.Port), 10)
 
 	volume.VolumeContext[vcSubsystemNQN] = data.SubsystemNQN
 	volume.VolumeContext[vcNamespaceID] = strconv.FormatUint(uint64(data.NamespaceID), 10)
 	volume.VolumeContext[vcNamespaceUUID] = data.NamespaceUUID
-	volume.VolumeContext[vcListenerAddress] = data.ListenerInfo.Address
-	volume.VolumeContext[vcListenerPort] = listenerPortStr
-	volume.VolumeContext[vcListenerHostname] = data.ListenerInfo.Hostname
 	volume.VolumeContext[vcGatewayAddress] = data.GatewayManagementInfo.Address
 	volume.VolumeContext[vcGatewayPort] = gatewayManagementInfoPortStr
+
+	// Store listeners as JSON
+	listenersJSON, err := json.Marshal(data.ListenerInfo)
+	if err != nil {
+		log.ErrorLog(context.TODO(), "Failed to marshal listener info: %v", err)
```

Comment:
Let the caller handle the logging. Logging with context.TODO will not help us in debugging as we will miss details about the RPC this error belongs to